### ISSUE TITLE
Commands and buttons no longer working

### ISSUE THREAD

**Issue by davidbrownell** [egodigital/vscode-powertools#22]
The following configuration was previously working, but it seems like recent updates to VSCode have introduced problems for powertool commands and buttons.

Repro steps:
Follow the instructions on the [powertools wiki](https://marketplace.visualstudio.com/items?itemName=ego-digital.vscode-powertools&ssr=false#buttons-) to create a button.

Expected:
A button appears on the toolbar.

Actual: 
No button appears (even after a reload).

VSCode:
```
Version: 1.33.1 (user setup)
Commit: 51b0b28134d51361cf996d2f0a1c698247aeabd8
Date: 2019-04-11T08:27:14.102Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.17763
```

Powertools:
```
Name: Power Tools
Id: ego-digital.vscode-powertools
Description: A set of useful and handy tools for Visual Studio Code.
Version: 0.40.0
Publisher: e.GO Digital GmbH
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ego-digital.vscode-powertools
```

Note that I originally discovered the problem when attempting to executed a command that had worked in earlier versions of VSCode and Powertools.

**Comment by ghost** [comment]
@davidbrownell 

Can you post an example configuration and code, which reproduces the issue(s)?

**Comment by davidbrownell** [comment]
Sure, although I copied it straight from the wiki.

settings.json:
```
"ego.power-tools": {
        "buttons": [
            {
                "text": "Click me!",
                "tooltip": "You run an awesome script by clicking that button!",
                "action": {
                    "type": "script",
                    "script": "my_button.js"
                }
            }
        ] 
}
```

my_button.js:
```
exports.execute = async (args) => {
    // args => https://egodigital.github.io/vscode-powertools/api/interfaces/_contracts_.buttonactionscriptarguments.html

    // s. https://code.visualstudio.com/api/references/vscode-api
    const vscode = args.require('vscode');

    vscode.window.showInformationMessage(
        'Hey, you clicked me!'
    );
};
```

**Comment by ghost** [comment]
@davidbrownell 

Hmm ... the example works for me: Button is visible, clickable and executing the code as expected.

Where are the files stored? In `.vscode` subfolder? Is that the only workspace/project, which is open?

**Comment by ghost** [comment]
I also tried it with Windows 10 (first one was Linux), without problems.

Is there something in your logs? (Command `Power Tools: Logs`?)

**Comment by davidbrownell** [comment]
Unfortunately, when I run the command ```Power Tools: Logs``` I see a popup that says ```No log files found!```

I tried a couple of additional things given that you weren't able to repro locally, but I still see the same results (no button). Here is what I tried:

- Disabled all other extensions
- Uninstalled and reinstalled Power Tools

**Comment by ghost** [comment]
Can you try to debug your script, as described here?: https://github.com/egodigital/vscode-powertools/issues/21#issuecomment-484711805

**Comment by davidbrownell** [comment]
It seems that there is something else going on. I started to follow the steps you mentioned and realized that I wasn't able to execute any Powertools commands.

When I select the VSCode Command ```Power Tools: Commands``` I see a popup that says ```No commands found!```.

To narrow down the problem, I have reduced my settings.json file to (this is all of it):
```
{
    "ego.power-tools": {
        "buttons": [
            {
                "text": "Click me!",
                "tooltip": "You run an awesome script by clicking that button!",
                "action": {
                    "type": "script",
                    "script": "my_button.js"
                }
            }
        ] ,
        "commands": {
            "DocumentItem": {
                "script": "DocumentItem.js"
            },
            "OSName": {
                "script": "OSName.js"
            }
        }
    }
}
```

**Comment by ghost** [comment]
Very strange ... when I use your settings, anything works fine on my machine: The button is visible and executable, like all commands ... I am currently testing on a Windows 10 machine and have the "same" VSCode environment:

```
Version: 1.33.1 (user setup)
Commit: 51b0b28134d51361cf996d2f0a1c698247aeabd8
Date: 2019-04-11T08:27:14.102Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.17763
```

I hope, that has nothing to do with Windows (Updates) ... yesterday I had to reinstall my system, because none of my Steam games could start anymore, after I had made a Windows update.

**Comment by ghost** [comment]
The only idea I currently have:

* Uninstall VSCode
* delete whole `.vscode` subfolder inside your user / home directory (`C:\Users\<USERNAME>\.vscode`)
* install VSCode again

**Comment by davidbrownell** [comment]
I went through the steps you mentioned but still see the same results after the VSCode reinstall (```No commands found!```).

**Comment by davidbrownell** [comment]
So, something changed and I now have access to the Power Tools commands as expected. Here is what I noticed....

I have a command that is used by a setting in another extension. The command is ```OSName``` and the setting is ```${workspaceRoot}/build/${command:OSName}/${buildType}```. I noticed that this command was working, indicating that VSCode was parsing the Power Tools commands in ```settings.json```.

I then modified ```OSName```'s JavaScript to verify that the setting was indeed referencing the output of the Power Tool command (it was).

From here, I ran ```Power Tools: Commands``` and saw all of my commands as expected. It seems that either accessing the command through the setting or editing the command's JavaScript perturbed enough to get things working again.

I apologize that I don't have more definitive information, but at least the issue is resolved!

Thank you for the help and prompt replies!

**Comment by davidbrownell** [comment]
Fortunately, I've been able to narrow this down. The Power Tools commands work with I have a folder open in VSCode. I see the behavior described above when a folder is not open.

Here are some scenarios that I tried:

- Open VSCode with a folder (Power Tools commands work as expected)
- Open VSCode without a file or folder (Power Tools commands do not work)
- Open VSCode with a file (Power Tools commands do not work)
- Open VSCode with a folder (works), close the folder in the same instance (does not work)
- Open VSCode without a file or folder (does not work), open a folder in the same instance (works)
